File: src/evai/command/CommandRegistry.java
package evai.command;

import java.util.HashMap;
import java.util.Map;

public class CommandRegistry {
    private static final Map<String, CommandHandler> handlers = new HashMap<>();

    @FunctionalInterface
    public interface CommandHandler {
        CommandResult handle(String action, String[] args);
    }

    public static void register() {
        handlers.put("help", new BaseCommands());
        handlers.put("status", new BaseCommands());
        handlers.put("whereami", new BaseCommands());
        handlers.put("clear", new BaseCommands());
        handlers.put("version", new BaseCommands());
        handlers.put("list", new InfoCommands());
        handlers.put("info", new InfoCommands());
        handlers.put("scan", new ScanCommands());
        handlers.put("check", new ScanCommands());
        handlers.put("map", new ScanCommands());
        handlers.put("debug", new DebugCommands());
        handlers.put("set", new SetCommands());
        handlers.put("nuke", new SetCommands());
        handlers.put("build", new BuildCommands());
        handlers.put("break", new BuildCommands());
        handlers.put("schem", new SchemCommands());
        handlers.put("spawn", new SpawnCommands());
        handlers.put("control", new SpawnCommands());
        handlers.put("kill", new SpawnCommands());
        handlers.put("listen", new EventCommands());
        handlers.put("fire", new EventCommands());
        handlers.put("js", new JsCommands());
        handlers.put("logic", new JsCommands());
        handlers.put("eval", new JsCommands());
        handlers.put("save", new SaveCommands());
        handlers.put("load", new SaveCommands());
        handlers.put("export", new ExportCommands());
    }

    public static CommandResult execute(String cmd) {
        if (cmd == null || cmd.trim().isEmpty()) return new CommandResult(false, "empty command");
        String[] parts = cmd.trim().split("\\s+");
        String action = parts[0];
        String[] args = new String[parts.length - 1];
        System.arraycopy(parts, 1, args, 0, args.length);
        CommandHandler handler = handlers.get(action);
        if (handler == null) return new CommandResult(false, "unknown command: " + action);
        return handler.handle(action, args);
    }
}

File: src/evai/command/CommandRegistry.kt
package evai.command

data class CommandResult(val ok: Boolean, val message: String = "", val data: Map<String, Any?>? = null) {
    fun toJson(): String = buildString {
        append("{\"ok\":$ok,")
        if (message.isNotBlank()) append("\"message\":${escape(message)},")
        if (data != null) {
            append("\"data\":{")
            append(data.entries.joinToString(",") { (k, v) -> "\"$k\":${toValue(v)}" })
            append("},")
        }
        append("\"timestamp\":${System.currentTimeMillis()}}")
    }

    private fun escape(s: String) = "\"${s.replace("\\", "\\\\").replace("\"", "\\\"").replace("\n", "\\n")}\""

    private fun toValue(v: Any?): String = when (v) {
        null -> "null"
        is String -> escape(v)
        is Number -> v.toString()
        is Boolean -> v.toString()
        is List<*> -> "[${v.joinToString(",") { toValue(it) }}]"
        is Map<*, *> -> "{${v.entries.joinToString(",") { (k, v) -> "\"$k\":${toValue(v)}" }}}"
        else -> escape(v.toString())
    }
}

object CommandRegistry {
    private val handlers = mutableMapOf<String, CommandHandler>()

    fun register() {
        handlers["help"] = BaseCommands()
        handlers["status"] = BaseCommands()
        handlers["whereami"] = BaseCommands()
        handlers["clear"] = BaseCommands()
        handlers["version"] = BaseCommands()
        handlers["list"] = InfoCommands()
        handlers["info"] = InfoCommands()
        handlers["scan"] = ScanCommands()
        handlers["check"] = ScanCommands()
        handlers["map"] = ScanCommands()
        handlers["debug"] = DebugCommands()
        handlers["set"] = SetCommands()
        handlers["nuke"] = SetCommands()
        handlers["build"] = BuildCommands()
        handlers["break"] = BuildCommands()
        handlers["schem"] = SchemCommands()
        handlers["spawn"] = SpawnCommands()
        handlers["control"] = SpawnCommands()
        handlers["kill"] = SpawnCommands()
        handlers["listen"] = EventCommands()
        handlers["fire"] = EventCommands()
        handlers["js"] = JsCommands()
        handlers["logic"] = JsCommands()
        handlers["eval"] = JsCommands()
        handlers["save"] = SaveCommands()
        handlers["load"] = SaveCommands()
        handlers["export"] = ExportCommands()
    }

    fun execute(cmd: String): CommandResult {
        val parts = cmd.trim().split("\\s+".toRegex())
        if (parts.isEmpty()) return CommandResult(false, "empty command")
        val action = parts[0]
        val args = parts.drop(1).toTypedArray()
        val handler = handlers[action] ?: return CommandResult(false, "unknown command: $action")
        return handler.handle(action, args)
    }

    interface CommandHandler {
        fun handle(action: String, args: Array<String>): CommandResult
    }
}